LINMA2380: Update exercise solutions for 2020--2021

Peiffap opened the pull request.

Jimvy commented:
In the main file, lines 16-18, a command `\exo` is defined to put stars in the exo title (for the difficulty I guess?).
Notwithstanding the fact that this is ugly, it also causes many "Token not allowed in a PDF string" errors due to the `\star` in math mode.
You can replace that with
```latex
\newcommand{\exo}[1]{\subsection{\texorpdfstring{%
\LARGE\forloop{stars}{0}{\value{stars} < #1}{$\star$}
}{%
\ifnum#1=1 #1 star \else #1 stars \fi%
}}}
```
to suppress the error.

If you know better LaTeX programming commands, you can use them of course. ^^

Jimvy commented on src/q7/matrix-INMA2380/exercises/ch1.tex:
```diff
@@ -205,256 +296,290 @@ \section{Matrix' algebra}
+(A \otimes B)^* = A^* \otimes B^*.
+\]
+
 \begin{solution}
   It is shown by
```
Add `\allowdisplaybreaks` between the lines so that `align` allows page breaks.

Peiffap replied:
I reworded it instead.

Jimvy commented on src/q7/matrix-INMA2380/exercises/matrix-INMA2380-exercises.tex:
```diff
@@ -16,7 +19,7 @@
 \usepackage{url}
 
```
Add
```
\vfuzz=318pt
```
so that exercise 1.8 stops giving `overfull \vbox` errors. The value 318 comes from the largest value given on the output, rounded towards +oo.

The problem is that the matrices are too large to fit within the previous page (where the statemen is located, page 8), so LaTeX puts the matrices on the next page (page 9), along with the single line of text above (in the same paragraph), and along with the `solution`'s upper curve. But this leaves a large amount of blank space on the bottom of page 8, exceeding LaTeX's acceptable level, and thus it cries.

In a sense, this is a "shut up" fix, but this is mostly the only reasonable fix for this warning, outside of rewording page 8 to fill in the blank space (either adding words or removing words).

Jimvy replied:
(However, the following warnings are present then:
```
Package mdframed Info: Box was splittet wrong
 starting loop to iterate the splitting point
(mdframed)              on input line 260.


Package mdframed Warning: correct box splittet fails
 It seems you are using a non splittable contents
(mdframed)                 on input line 260.

Package mdframed Info: You first box width is to small
 mdframed fixed it
(mdframed)              on input line 260.
[7]
```
Mostly because `mdframed` still has difficulty of letting this go. So we traded an `overfull \vbox` for a `mdframed` warning...)

Peiffap replied:
Reworded, no longer a problem.

Jimvy commented on src/q7/matrix-INMA2380/exercises/ch1.tex:
```diff
@@ -110,92 +201,92 @@ \section{Matrix' algebra}
 
 \begin{solution}
-  Let $T,U$ be upper triangular Toeplitz,
+  Let $T,U$ be upper triangular Toeplitz matrices.
+  Then
```
If you don't want to put such a large `\vfuzz` value, you can try creating 2 paragraphs here, so that the solution header appears on PDF page 8 (well, 7 not counting the front page). But it looks kinda ugly ^^

Jimvy replied:
Correction: you can then even remove entirely the `vfuzz=`. But I still find it ugly.

Jimvy reviewed (commented):
Aside from this, nothing to say. I haven't checked the math so I assume that it is OK ;)

Jimvy commented:
En fait, j'ai un autre soucis : chez toi, quand tu compiles via le Makefile, ça compile sans problème les deux documents (version sans et avec sol) ?
Chez moi, la version avec Sol compile, la version sans Sol ne compile pas, avec un message d'erreur sur le `comment` "Bad space factor (0)". Et je ne comprends absolument pas d'où ça peut sortir...

Jimvy commented:
Ok, visiblement ce block `comment` est assez fragile niveau retours à la ligne etc, et lorsqu'on l'inclut dans l'environnement solution qui est parfois réduit à juste un `\expandafter\comment \expandafter\endcomment`, ça a vraiment l'air de péter le truc.
Vu que les anciennes réponses sont dans l'historique, ça va si on enlève simplement les blocs de commentaire ? :)

Peiffap commented:
I updated the `/exo` command and removed the comments. Everything compiles on my machine, including from `make pdf`. I also don't have any errors/warnings other than the perennial "You have requested ..." ones we can't get rid of as far as I know.
